src/components/WalletModal/index.tsx

```diff
@@ -113,20 +121,46 @@ const WALLET_VIEWS = {
   PENDING: 'pending',
 }
 
+const sendAnalyticsWalletBalanceUserInfo = (
+  balances: (CurrencyAmount<Currency> | undefined)[],
+  nativeCurrencyBalanceUsd: number
+) => {
+  const walletTokensSymbols: string[] = []
+  const walletTokensAddresses: string[] = []
+  balances.forEach((currencyAmount) => {
+    if (currencyAmount !== undefined) {
+      const tokenBalanceAmount = formatToDecimal(currencyAmount, currencyAmount.currency.decimals)
+      if (tokenBalanceAmount > 0) {
+        const tokenAddress = getTokenAddress(currencyAmount.currency)
+        walletTokensAddresses.push(getTokenAddress(currencyAmount.currency))
+        walletTokensSymbols.push(currencyAmount.currency.symbol ?? '')
+        const tokenPrefix = currencyAmount.currency.symbol ?? tokenAddress
+        user.set(`${tokenPrefix}${CUSTOM_USER_PROPERTIES.WALLET_TOKEN_AMOUNT_SUFFIX}`, tokenBalanceAmount)
```

Comment:
oh i see... hm i'm wondering what a better way to do this would be. maybe we can just send a map of all the balances instead of individual user properties?

also, if it's a user properly can we still look at historical values?

Reply:
objects are generally really not preferrable in amplitude. it hinders a lot of charting functions that would otherwise be available. also this one in particular was a direct recommendation from amplitude folks, i'll forward you the email xchange. 

we can look at historical values because we can look at historical events, and all the user properties at that time will be attacahed to the event.